### PR TITLE
Opal 734/807

### PR DESCRIPTION
**Pull request by EF-USAF**
Includes prometheus setup and revamped opal install guide

### CODE REVIEW

**Review by MM-USAF — DISMISSED** [review]

File: opal/aks-values.yaml [hunk 1]
```diff
@@ -225,6 +225,10 @@ nginx:
   enabled: true
   appValues:
     namespace: *opalNS
+    condaChannelImage:
```
Comment (hackowitz-af) [inline opal/aks-values.yaml:228]: Should these changes be included in this PR? It seems unrelated to the README

File: README.md [hunk 2]
```diff
@@ -1,98 +1,170 @@
...
 
->$helm install argoregcred . --create-namespace -n argocd
+##### Setting up ArgoCD regcred
+
+change directory to the regcred-init directory and run the following command:
+
+``` bash
+helm install argo-regcred . --create-namespace -n argocd
+```
+
```
Comment (hackowitz-af) [inline README.md:113]: Where is the `regcred-init` directory?

File: README.md [hunk 3]
```diff
@@ -1,98 +1,170 @@
...
+##### Install ArgoCD helm chart
 
-### Install ArgoCD helm chart
 now that the argoCD values are configured properly for a local deployment, helm install the argocd from the parent directory of ArgoCD repository
 
->$helm install argo ./chart -n argocd
+``` bash
+cd argo/chart
```
Comment (hackowitz-af) [inline README.md:119]: I found this path ambiguous

File: README.md [hunk 4]
```diff
@@ -1,98 +1,170 @@
...
+
+
+Access the argoCD interface by opening a browser and navigating to the following:
+
 > localhost:8080
 
-there should be a login page for argoCD
+This should take you to the login page for argoCD
+
 >username: **admin**
```
Comment (hackowitz-af) [inline README.md:160]: This formatting is inconsistent with similar text elsewhere in the document. Preferred might look like

Username:
> admin

Password:
> output from the last section (remove the trailing '%' symbol if present)

File: README.md [hunk 5]
```diff
@@ -104,70 +176,132 @@ Username:
 Password:
 > enter your github token
 
-Click the "*connect*" button at the top of the page to save the connection
+Click the `Connect` button at the top of the page to save the connection
 
-If the information was entered correctly the connection status will be *Successful*
+If configured correctly, the connection status will show as *Successful*
```
Comment (hackowitz-af) [inline README.md:182]: The confluence page has something here about enabling argo ingress. Is this desired to keep? The text is


The argo ingress is not enabled by default, to change this, navigate to argo/chart/values.yaml and modify the following lines (Note, if you're using vim, an easy way to navigate to these lines is :<line number> )

At (or around) line 2017, change **"enabled: false"** to **"enabled: true"**

At (or around) line 2033, change - **[]** under "hosts: " to - **argo.10.96.30.9.nip.io**

File: README.md [hunk 6]
```diff
@@ -104,70 +176,132 @@ Username:
...
 
 ## Deploying OPAL
-With argoCD running and configured properly, opal applications can be helm installed.
+With argoCD running and configured, the opal helm chart can now be installed.
+
```
Comment (hackowitz-af) [inline README.md:185]: The confluence page says to close down the port forwarding from above here. Is this necessary?

File: README.md [hunk 7]
```diff
@@ -104,70 +176,132 @@ Username:
...
-**NOTE**: A github account and access token with repo scope is currently required to use this functionality.
+
+> **NOTE**: Some repositories may require a github account and access token with repo scope.
 
 To use git-sync, place your token in opal-setup, and rename it to `git-creds`. In opal-setup/values, fill in your username in jupyterhub:appValues:gitSync
 
-### Helm install OPAL
+### Install the OPAL helm chart
+
+Prior to installing OPAL, you will likely need to change your working branch in git. At the time of writing, the current production branch is `aks-comp-rebase-test`. If you are testing a specific branch or have configurations stored in another branch, use that instead. Additionally, ensure that the value of `targetRevision` in `opal-helm/opal-setup/values.yaml` is set to the correct branch, or you may run into unforeseen issues.
```
Comment (hackowitz-af) [inline README.md:230]: I'm cautious that: the production version:
-  is a branch not a tag
- is not the main, master or vX.Y branch
- is named  `...-test`.

If this is correct, I would think it needs explaining here vs. just saying to use that branch

File: README.md (hunk 2 above)
Comment (hackowitz-af) [reply README.md:113]: Seems to be in a different git repo, but is not mentioned

Comment (EF-USAF) [reply README.md:113]: Part of k8s-utils, mentioned in prereqs

File: README.md (hunk 5 above)
Comment (EF-USAF) [reply README.md:182]: yeah that's a bad habit that I didn't quash soon enough. best practice is to only expose argo over port-forward

File: README.md (hunk 6 above)
Comment (EF-USAF) [reply README.md:185]: Not usually. Some people have had issues with accessing any of the ingress points if they leave the port-forward open. I've never seen it happen personally

File: README.md (hunk 7 above)
Comment (EF-USAF) [reply README.md:230]: This branch got so far ahead of main and had so many differing implementations at one point that it was easier to just use this as the new main. We now have a live environment running off of it and I don't want anything to break, so it just stuck like that.

This chart is also still in its infancy. There aren't official releases, and the live server exists as a proof of concept that largely goes unused.

File: README.md [hunk 8]
```diff
@@ -104,70 +176,132 @@ Username:
...
+To allow kubernetes to authenticate with a private image registry, you'll first need to log in to that registry through the docker cli.
+In the case of IronBank, this will be `docker login registry1.dso.mil`. If using a different private registry, use that URL instead.
 
-### Add docker configuration JSON
-In order for the helm install to work properly in a local instance, it is necessary to add the docker config json file in the opal-helm/opal-setup directory.  This is needed for the helm chart to pull the appropriate image from a private repository.
+Upon successful login, assuming you are using a Linux system, the login command should have created the file `~/.docker/config.json`. Copy this to `opal-helm/opal-setup`, and rename it to `dockerconfig.json`
 
 ### Git Sync
+
 If enabled in the values.yaml, jupyterhub will add sidecars containers to the hub pod that will periodically pull in updates from the `opal` and `weave` repositories.
```
[the remark this answers is not in the document]
Comment (EF-USAF) [reply README.md:222]: These sidecars are specifically part of the Jupyterhub helm chart. Argo only installs the charts.

File: opal/aks-values.yaml (hunk 1 above)
Comment (EF-USAF) [reply opal/aks-values.yaml:228]: I am not making a PR for a four-line fix. Especially not when the mandatory in-office policy means my PRs can only be reviewed on M/F. By condensing changes, I can get more updates through in a timely manner.

File: README.md (hunk 3 above)
Comment (EF-USAF) [reply README.md:119]: How so? The prerequisites have you clone down a repo called `argo`.

File: README.md (hunk 7 above)
Comment (EF-USAF) [reply README.md:230]: I do plan to eventually get this all into the actual main branch, but it's unfortunately quite low in priorities over our other tasking

Comment (EF-USAF) [reply README.md:230]: I guess ignore the previous comment. I've archived the old main and copied/renamed rebase to make things less confusing

**Review by MM-USAF — DISMISSED** [review]

File: README.md (hunk 3 above)
Comment (hackowitz-af) [reply README.md]: I think my difficulty stems from the frequent `cd`s, together with the fact I am mainly using `ls` to track the contents of each directory. So it just takes a little memory/backtracking to realize `argo/chart` is `../../argo/` or `~/309thEDDGE/argo/chart` in my case, not just `./argo/chart/`.

The latest change fixes it, thanks.

File: opal/aks-values.yaml (hunk 1 above)
Comment (hackowitz-af) [reply opal/aks-values.yaml:228]: Copy - yeah it does seem the process/policy interferes with work being done quickly, or for small changes to ever happen. Whatever gets the code released works for me; just doing my due diligence and this stood out to me

**Review by hackowitz-af — APPROVED** [review]
All comments resolved, approved! Thanks for bearing with me learning Ethan

File: README.md [hunk 9]
```diff
@@ -1,173 +1,361 @@
...
+```
 
-Access the argoCD interface by opening a browser and navigate to the following:
-> localhost:8080
+##### Install ArgoCD helm chart
 
-there should be a login page for argoCD
->username: **admin**
->password: output from the last section (remove the trailing '%' symbol if present)
+Now that the registry credentials have been configured, `cd` into the `argo` repository cloned previously, and install the helm chart:
```
Comment (JN-USAF) [inline README.md:121]: Specify that when running the 'helm install argo . -n argocd' command, you should be in the 'argo/chart' directory.

**Review by JN-USAF — APPROVED** [review]
I was able to follow the README and get everything deployed locally

**Review by MM-USAF — APPROVED** [review]
Great write up for the readme

**Pull request merged**